client/src/components/Ticketing/ticketingmanager/SubscriptionRedemption/SubscriptionTicketItemRow.tsx

```diff
@@ -0,0 +1,289 @@
+import React from 'react';
+import {Field, FormikErrors, useFormikContext} from 'formik';
+import {OptionRowProps} from '../Event/components/OptionUpdateTable';
+import {OptionSelect} from '../Event/components/OptionSelect';
+import {FormButton} from '../Event/components/FormButton';
+import {TrashCanIcon} from '../../Icons';
+import {Event} from './SubscriptionRedemptionUtils';
+import {getDate} from '../Event/components/ShowingUtils';
+import format from 'date-fns/format';
+import {SubscriptionRedemptionFormRow} from './SubscriptionTicketItemUpdateTable';
+
+const getHandleEventChangeAdmin =
+  (
+    options: Event[],
+    setOptions: (value) => void,
+    parentFieldName: string,
+    setFieldValue: (
+      name: string,
+      value: any,
+    ) => Promise<void | FormikErrors<unknown>>,
+    currentValue: SubscriptionRedemptionFormRow,
+  ) =>
+  async (inputEvent) => {
+    const newEventId = +inputEvent.target.value;
+    setOptions(
```

Comment:
Is there any way that this block can be written in a bit of a more readable fashion, rather than very nested ternaries.

Reply:
Updated